docs/angular/rxobject.md

```diff
@@ -0,0 +1,416 @@
+
+## Update Data Using `RxObject`
+
+If you have explored the Coworking feature of the CSXL Application, you likely know that the data on the Coworking page *updates every 5 seconds*. Say that `pollStatus()` is a function that exists *within the `CoworkingService`* that runs every 5 seconds to update the current status of the CSXL and coworking data. Let's take a look here:
+
+**In `coworking.service.ts`:**
+
+```ts
+@Injectable({
+  providedIn: 'root'
+})
+export class CoworkingService {
+
+  // OTHER FUNCTIONALITY HIDDEN
+
+  /** This method is called every 5 seconds to update the coworking data. */
+  pollStatus(): void {
+    this.http.get<CoworkingStatus>('/api/coworking/status').subscribe((status) => {
+      console.log("FETCHED STATUS! Now what?");
+    });
+  }
+}
+```
+
+How would we go about storing our data so that we can access this data in our component? Let's use `RxObject`!
+
+First, as you remember, `RxObject` is an *abstract class* - therefore, it cannot be instantiated directly. We need to create a *subclass* for `RxObject` that pertains to our specific feature and the data we are trying to store. In this case, we are trying to store an object of type `CoworkingStatus`. So, let's define a new class - `RxCoworkingStatus` - that extends `RxObject`! We can define this using the code below in a new file.
+
+**In `rx-coworking-status.ts`**
+
+```ts
+export class RxCoworkingStatus extends RxObject<CoworkingStatus> {}
+```
+
+Notice that this is a one-liner! We do not need to override or overload any methods. Notice also though that the *generic type* goes away - this is because we extend `RxObject<CoworkingStatus>`, which fills in this generic type!
+
+Now, ***in our service***, we can create two new fields - one to store our `RxCoworkingObject` which allows us to dynamically update our data, and then one public observable for our components to use that is automatically updated with new data! The fields are shown below:
+
+```ts
+private status: RxCoworkingStatus = new RxCoworkingStatus();
+public status$: Observable<CoworkingStatus> = this.status.value$;
+```
+
+Notice that `status`, which is our `RxCoworkingStatus` object, is private - this is because we want to abstract all of this functionality and keep it from our Components. Instead, the only thing accessible to the components should be the *observable* that is automatically updating with new data, `status$`! Notice that this observable is simply the public `.value$` observable from our `status` object that we defined in the previous part.
+
+Finishing this functionality is easy! All we need to do now is call the `.set()` method on our `RxCoworkingStatus` object (remember, this was *inherited* from `RxObject` because `RxCoworkingStatus` extends `RxObject`). We can do this in `pollStatus()`, which is supposed to update our data! Look at the following final construction:
```

Comment:
Seeing these comments makes me wonder if a refactoring of RxObject is in store for simplifying things 👀 

This may be a winter break goal, but this has me wondering: _why doesn't RxObject implement `Observable<T>`_ and then we could get rid of this ugliness and just make `public status$: RxCoworkingStatus` without the private side. Now, this would mean that the public methods of `RxCoworkingStatus` are exposed outside of the service, but this might be an OK trade-off. This extra step of doing public/private _again_ feels like it's a bit overkill and would love to simplify what the service-layer implementation looks like.

As I write this out, I am also wondering if all we really want is a subclass of `Subject` that is effectively `ReplaySubject(1)` where we can directly access the last value in subclasses without having to subscribe. This is a path worth exploring at some point soon.